### DISCUSSION TITLE
Access x-for keys or single key?

### DISCUSSION THREAD

**Discussion by ryatkins** [alpinejs/alpine#960] [1. Help]
Is there a way to access the bound key for each element in a x-for? 

I wish to iterate over all keys or access a single key.   

Of course it is the same as my index put accessing it via an attribute will help with a 3rd party library I’m trying to implement.

**Comment by KevinBatdorf** [comment]
You might be able to use `__x_for_key` See here: https://github.com/alpinejs/alpine/blob/master/src/directives/for.js
